### PR TITLE
Mvdb/fix encoding issue

### LINKED ISSUE #55: Encoding issue when calling API
Hi there,

We encountered encoding issues when trying to send Swedish or French text. We might want to adapt [the following code](https://github.com/facebook/facebook-python-ads-sdk/blob/master/facebookads/api.py#L386-L387):
```
        if params:
            params = _top_level_param_json_encode(params)
            keyvals = ['%s=%s' % (key, urllib.parse.quote(value))
                       for key, value in params.items()]
            call['body'] = '&'.join(keyvals)
```

At first, we thought (as always with encoding) that we should do it ourselves before handing data to the Api, but the issue is that when reading data, the api will return `unicode` object instead of `basestring` object, hence it makes it difficult to handle every situation. For instance:
```
data = {...}
data = {key: value.encode("utf-8") for key, value in data.iteritems()}
creative = AdCreative().remote_create(params=data)  # this one will work
creative.remote_read(fields=["name", ...])  # let's say the name contains utf-8
adgroup_name = "Adgroup for " + creative["name"]  # here we have utf-8
adgroup_data = {"name": "adgroup_name", ...}
adgroup = AdGroup().remote_create(params=adgroup_data)  # this call will fail
```
The exception will be `KeyError: u'\xe9'` when calling the `urllib.parse.quote` method.

The fix is fairly easy and easily testable, let me know if you're interested in a PR, I can provide it quickly

### PR DESCRIPTION
**Pull request by m-vdb**
fixes #55

### CODE REVIEW

**Comment by m-vdb** [comment]
there you go @ebzlo

**Comment by m-vdb** [comment]
@ebzlo gentle ping :)

**Comment by ebzlo** [comment]
Hey @m-vdb , sorry for delay. Can you squash and rebase? I will pull in for review.

**Comment by m-vdb** [comment]
@ebzlo you mean you prefer one commit?

**Comment by ebzlo** [comment]
@m-vdb yup

**Comment by agriffis** [comment]
@m-vdb Did you see this is failing in Travis on Python 3?

**Comment by m-vdb** [comment]
thanks @agriffis, I'll fix that. I'm not very familiar with (and fan of) Python 3 :)

File: facebookads/utils/urls.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    to utf-8 before using `urllib.parse.quote`.
+    In case it's not a string (an int for instance),
+    we still try to convert it.
+
+    Args:
+            string: The string to be properly encoded.
+    """
+    # handle other stuff than strings
+    if not isinstance(string, six.string_types):
+        string = six.text_type(string).encode('utf-8') if six.PY3 else bytes(string)
```
Comment (m-vdb) [inline facebookads/utils/urls.py]: inspired by [Django's encoding methods](https://github.com/django/django/blob/1.6.11/django/utils/encoding.py)

**Comment by m-vdb** [comment]
@ebzlo @agriffis all good. I rebase and tests now pass on all envs.

**Comment by agriffis** [comment]
I'm personally not a fan of accepting types other than strings, since many types can be stringified but this won't result in correct behavior (dicts, lists, objects, etc).  I'd suggest dropping that part and keeping the API strict, something like this:
```
if not isinstance(string, six.string_types):
    raise ValueError("Non-string passed to quote_with_encoding")
if not isinstance(string, six.binary_type):
    string = string.encode('utf-8')
```

If you really want to convert ints, then recognize them explicitly with `six.integer_types`

**Comment by m-vdb** [comment]
agree that we cannot support any type, but integer are more frequent than string in FB API ><. I'll check for `integer_types` and raise in case of any other type. Is this ok?

**Comment by agriffis** [comment]
@m-vdb Sounds good to me, though @ebzlo is the gatekeeper. I'm just an opinionated third party. ;-)

**Comment by m-vdb** [comment]
all good @ebzlo

**Comment by ebzlo** [comment]
Looks good. @m-vdb 

@agriffis FWIW I appreciate the opinions :)

Pulling in for review!

**Comment by ebzlo** [comment]
Hey @m-vdb we've accepted the diff for review, but I couldn't get this merged.
I've credited you on the releases page now that 2.4 is live :)
https://github.com/facebook/facebook-python-ads-sdk/releases/tag/2.4.0

**Comment by m-vdb** [comment]
you rock @ebzlo ! thanks for the credit too :heart:

**Pull request closed without merging**